shared/include/userver/utils/fixed_array.hpp

```diff
@@ -63,18 +77,29 @@ class FixedArray final {
   const T* cend() const noexcept { return data() + size_; }
 
  private:
+  T* NonEmptyData() noexcept {
+    UASSERT(size_);
+    return data();
+  }
+
+  const T* NonEmptyData() const noexcept {
+    UASSERT(size_);
+    return data();
+  }
+
   T* storage_;
```

Comment:
How about 
```suggestion
  T* storage_{nullptr};
  std::size_t size_{0};
```
This my simplify the constructors